src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -337,12 +410,26 @@ internal sealed class SrCnnEntireModeler
             private double[] _trends;
             private double[] _curWindow;
 
-            public SrCnnEntireModeler(double threshold, double sensitivity, SrCnnDetectMode detectMode)
+            public SrCnnEntireModeler(double threshold, double sensitivity, SrCnnDetectMode detectMode, int period, SrCnnDeseasonalityMode deseasonalityMode)
             {
                 _threshold = threshold;
                 _sensitivity = sensitivity;
                 _detectMode = detectMode;
+                _period = period;
                 _predictArray = new double[_lookaheadWindowSize + 1];
+
+                if (deseasonalityMode == SrCnnDeseasonalityMode.Stl)
+                {
+                    _deseasonalityFunction = new StlDeseasonality();
+                }
+                else if (deseasonalityMode == SrCnnDeseasonalityMode.Mean)
+                {
+                    _deseasonalityFunction = new MeanDeseasonality();
+                }
+                else // if (deseasonalityMode == SrCnnDeseasonalityMode.Median)
+                {
+                    _deseasonalityFunction = new MedianDeseasonality();
+                }
```

Comment:
This can be changed to a `switch` statement.

Reply:
In the default case, instead of writing `deseasonalityMode == SrCnnDeseasonalityMode.Median` you can add
```
Contracts.Assert(deseasonalityMode == SrCnnDeseasonalityMode.Median);
```
(that way, if in the future a value is added to the `SrCnnDeseasonalityMode` enum, it will be easier to detect that this code needs to be updated).


---
In reply to: [443404193](https://github.com/dotnet/machinelearning/pull/5202#discussion_r443404193) [](ancestors = 443404193)